### PR TITLE
Frontend: Refactor layout

### LINKED ISSUE #53: Improvement: element positioning on page
# Description
Elements are currently hard coded in their positioning which prevents the implementation of new panels (like the shortcut panel) 
- Width of the different panels needs to be consistent between each other
- Graph should be visible under the panel
- Graph should be interactive on all areas that are not covered by panels (beside gap between panels) 
- When too many panels are on screen to be displayed, it should be possible to scroll to see the hidden panels

### PR DESCRIPTION
**Pull request by alrapal**
# Description
- Removed sizing from HTML element to use div as container for more consistency
- Removed the deactivation of the legend when option where toggled on and vice versa
- Panels will stack on each other and when screen height is overflowing, scrolling will be activated
- Interaction with the graph are not prevented if a panel is activated (around the panel)
- Graph is visible under panel container
# Screenshots
## Scroll bar in panel space (smaller screen)
Screenshot 2022-12-15 at 18 28 48

## Interaction with graph and width consistency (bigger screen) 
Screenshot 2022-12-15 at 18 31 29

closes #53

### CODE REVIEW

**Comment by RuthgerD** [comment]
Shoudnt the query panel stay a the the bottom?

**Comment by alrapal** [comment]
> Shoudnt the query panel stay a the the bottom?

Do you mean it should always be last in the list of panels? Because it is atm
But if you are talking about a position fixed, that is exactly why this PR is made for, get rid of all fixed position (which makes very difficult to implement new panels). 
But I am open to any insights if you have an idea

**Comment by RuthgerD** [comment]
No I meant the query panel should be fixed at the bottom, if its just a floating panel then it would be jumping all over the place as you toggle some other panels.

**Review by ktpa — CHANGES_REQUESTED** [review]
Suggestions:

image
Set icons to match the order of which the features are displayed, in falling order:
- Legend
- Graph options
- Filter options

Remove trailing ":" on Filter Options for consistency.

Responsive sidebar panels works as expected, well done.

Might be a good idea to consider the @RuthgerD suggestion and see how that could be implemented. I would not say that it is part of this refactor PR though.

**Comment by alrapal** [comment]
> No I meant the query panel should be fixed at the bottom, if its just a floating panel then it would be jumping all over the place as you toggle some other panels.

Mmmh. This is basically the previous implementation which leads to a lot of problems when implementing other panels (like shortcut panels) 
Having the panel fixed is taking it out of the flex container basically. IT makes the UI very difficult to extend and improve. 
Also, how would you handle the different screen sizes? Having the panels floating with one fixed? I foresee a lot of issues with this kind of implementation. 
We can have a chat on zoom if you want to iterate through ideas but as far as now, I don't see how this would be good implementation as suggested now :x

**Comment by RuthgerD** [comment]
Flex allows you to put things at the end of the container, dont really see the issue, as long as the container holding the panels is full height.

**Comment by alrapal** [comment]
> Flex allows you to put things at the end of the container, dont really see the issue, as long as the container holding the panels is full height.

That for sure, but it will still be pushed down if overflows. But if that is okay, then I can for sure implement that =)

**Comment by RuthgerD** [comment]
>  But if that is okay

yes, dont spend too much time on supporting small screens, the common case needs to be solid first.

**Comment by alrapal** [comment]
> > But if that is okay
> 
> yes, dont spend too much time on supporting small screens, the common case needs to be solid first.

Some updates:
I played around with the code to try some different solutions and we have some trade off to make if we go for this (at this stage)
- If we make the container full screen to put the filter at the bottom, we have to activate interaction with the container in order to overflow (which is necessary for adding new panels). This will prevent interaction with the graph (like hovering feature, and all) on the area covered by the container even if no panels are open. I personally think that the benefit to have a fixed position for the filter is not worth losing interaction with the graph on that area. (cf screenshot)
Screenshot 2022-12-19 at 17 17 33

Any insights are welcome, but I don't see how to fix this without reworking the whole frontend structure (which is out of the scope of this PR, since this is to allow more panels to be added in a smooth way). 
I suggest we make an issue for this and try to address it once the rework on the panels as been done.

**Comment by alrapal** [comment]
With design suggested by Ruthger
Screenshot 2022-12-22 at 14 22 09

**Comment by alrapal** [comment]
Btw, we got feedback about the opacity of the links and they would like to have it at 0.3 instead of 0.15. I included this into this PR.

**Comment by RuthgerD** [comment]
lgtm

File: frontend/src/components/Panel.svelte [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
     //Boolean variables
     export let show_legend_placeholder: boolean; 
-    export let show_menu_placeholder: boolean; 
+    /* export let show_settings_placeholder: boolean; */
```
Comment (ktpa) [inline frontend/src/components/Panel.svelte:12]: Any particular reason to leave this here?

File: frontend/src/components/SideBar.svelte [hunk 2]
```diff
@@ -1,34 +1,23 @@
 <script lang="ts">
   import SvgButton from "./SvgButton.svelte";
+  import ButtonSeparator from "./Sidebar/ButtonSeparator.svelte";
 
-   export let show_menu: boolean;
+   export let show_settings: boolean;
```
Comment (ktpa) [inline frontend/src/components/SideBar.svelte:5]: Great change, reflects what is actually happening a lot better.

File: frontend/src/components/utils/HeaderWithCloseButton.svelte [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+    import { createEventDispatcher } from 'svelte';
+    const dispatch = createEventDispatcher();
```
Comment (ktpa) [inline frontend/src/components/utils/HeaderWithCloseButton.svelte:3]: dispatch is missing a type

**Review by ktpa — CHANGES_REQUESTED** [review]
Nice additions and changes but there are still a number of issues that needs to be addressed, namely:

- Ubuntu 20.04 Chrome looks like below:
Screenshot from 2022-12-22 18-30-50
Is this how it is supposed to look, with a scroll bar?


- Non-interactive mode stops when you change settings, intentional perhaps?
- I can't seem to see or locate the shortcut section of the settings panel. Was this part of this PR?
- Firefox complains that the page seems to be slowing down its performance. Not sure if this is exclusive to the refactor but possibly not. Same issues except for performance hit persist over Chrome/Firefox.

**Comment by alrapal** [comment]
> Nice additions and changes but there are still a number of issues that needs to be addressed, namely:
> 
> * Ubuntu 20.04 Chrome looks like below:
>   Screenshot from 2022-12-22 18-30-50
>   Is this how it is supposed to look, with a scroll bar?
  - Yes, the graph options has its own size. We would need to refactor that to fix this. But the scroll is the expected behaviour
> * Non-interactive mode stops when you change settings, intentional perhaps?
  - Mmmmh, not sure. On the other hand, being able to change settings on the graph with a graph moving can be difficult to actually see results, so I would consider this as beneficial. 
> * I can't seem to see or locate the shortcut section of the settings panel. Was this part of this PR?
 - No, separate PR for shortcut panel and shortcuts
> * Firefox complains that the page seems to be slowing down its performance. Not sure if this is exclusive to the refactor but possibly not. Same issues except for performance hit persist over Chrome/Firefox.
 - I had the same warning in firefox prior to starting working on this refactoring. 

Safari and chrome do work fine. Also fixed the bug on safari that was preventing displaying the filter panel (thanks to refactoring layout, was not willingly haha)

File: frontend/src/components/Panel.svelte (hunk 1 above)
Comment (alrapal) [reply frontend/src/components/Panel.svelte:12]: No, forgot to remove! Will fix it!

File: frontend/src/components/utils/HeaderWithCloseButton.svelte (hunk 3 above)
Comment (alrapal) [reply frontend/src/components/utils/HeaderWithCloseButton.svelte:3]: createEventDispatcher is a function imported from svelte library. Don't really know what time to put. Function?

**Review by ktpa — APPROVED** [review]
lgtm

**Pull request merged**